File: WRUW/AboutViewController.m
//
//  AboutViewController.m
//  WRUW
//
//  Created by Nick Jordan on 4/13/15.
//  Copyright (c) 2015 Nick Jordan. All rights reserved.
//

#import "AboutViewController.h"
#import "WRUWModule-Swift.h"

@interface AboutViewController ()

@end

@implementation AboutViewController
@synthesize appInfoText, wruwInfoText;

- (void)viewDidLoad {
    [super viewDidLoad];
    
    [self setupTextView:appInfoText];
    [self setupTextView:wruwInfoText];
}

- (void)setupTextView:(UITextView *)textView {
    [textView setUserInteractionEnabled:YES];
    [textView setEditable:NO];
    [textView setDataDetectorTypes:UIDataDetectorTypeAll];
    [textView setSelectable:YES];

    [textView setLinkTextAttributes:@{
        NSForegroundColorAttributeName: [[ThemeManager current] wruwMainOrangeColor],
        NSUnderlineStyleAttributeName: [NSNumber numberWithInt:NSUnderlineStyleSingle]
    }];
}

@end

File: WRUW/View Controllers/AboutViewController.swift
import UIKit

class AboutViewController: UIViewController {

    @IBOutlet weak var wruwInfoText: UITextView!
    @IBOutlet weak var appInfoText: UITextView!

    override func viewDidLoad() {
        super.viewDidLoad()

        setupTextView(textView: wruwInfoText)
        setupTextView(textView: appInfoText)
    }

    func setupTextView(textView: UITextView) {
        textView.isUserInteractionEnabled = true
        textView.isEditable = false
        textView.dataDetectorTypes = .all
        textView.isSelectable = true

        textView.linkTextAttributes = [
            .foregroundColor: ThemeManager.current().wruwMainOrangeColor,
            .underlineStyle: NSUnderlineStyle.single.rawValue
        ]
    }
}
